pkg/controller/logstash/pod_test.go

```diff
@@ -0,0 +1,205 @@
+				assert.Equal(t, corev1.ResourceRequirements{
+					Limits: map[corev1.ResourceName]resource.Quantity{
+						corev1.ResourceMemory: resource.MustParse("3Gi"),
+					},
+				}, GetLogstashContainer(pod.Spec).Resources)
+			},
+		},
+		{
+			name: "with user-provided init containers",
+			logstash: logstashv1alpha1.Logstash{Spec: logstashv1alpha1.LogstashSpec{
+				PodTemplate: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						InitContainers: []corev1.Container{
+							{
+								Name: "user-init-container",
+							},
+						},
+					},
+				},
+			}},
+			assertions: func(pod corev1.PodTemplateSpec) {
+				assert.Len(t, pod.Spec.InitContainers, 1)
+				assert.Equal(t, pod.Spec.Containers[0].Image, pod.Spec.InitContainers[0].Image)
+			},
+		},
+		{
+			name: "with user-provided labels",
+			logstash: logstashv1alpha1.Logstash{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "logstash-name",
+				},
+				Spec: logstashv1alpha1.LogstashSpec{
+					PodTemplate: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"label1":      "value1",
+								"label2":      "value2",
+								NameLabelName: "overridden-logstash-name",
+							},
+						},
+					},
+					Version: "8.6.1",
+				}},
+			assertions: func(pod corev1.PodTemplateSpec) {
+				labels := (&logstashv1alpha1.Logstash{ObjectMeta: metav1.ObjectMeta{Name: "logstash-name"}}).GetIdentityLabels()
+				labels[VersionLabelName] = "8.6.1"
+				labels["label1"] = "value1"
+				labels["label2"] = "value2"
+				labels[NameLabelName] = "overridden-logstash-name"
+				assert.Equal(t, labels, pod.Labels)
+			},
+		},
+		{
+			name: "with user-provided ENV variable",
+			logstash: logstashv1alpha1.Logstash{Spec: logstashv1alpha1.LogstashSpec{
+				PodTemplate: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "logstash",
+								Env: []corev1.EnvVar{
+									{
+										Name:  "user-env",
+										Value: "user-env-value",
+									},
+								},
+							},
+						},
+					},
+				},
+			}},
+			assertions: func(pod corev1.PodTemplateSpec) {
+				assert.Len(t, GetLogstashContainer(pod.Spec).Env, 1)
+			},
+		},
+		{
+			name: "with user-provided volumes and volume mounts",
+			logstash: logstashv1alpha1.Logstash{Spec: logstashv1alpha1.LogstashSpec{
+				PodTemplate: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "logstash",
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										Name: "user-volume-mount",
+									},
+								},
+							},
+						},
+						Volumes: []corev1.Volume{
+							{
+								Name: "user-volume",
+							},
+						},
+					},
+				},
+			}},
+			assertions: func(pod corev1.PodTemplateSpec) {
+				assert.Len(t, pod.Spec.Volumes, 2)
+				assert.Len(t, GetLogstashContainer(pod.Spec).VolumeMounts, 2)
+			},
+		},
+	}
+	for _, tt := range tests {
```

Comment:
Can we add test coverage for custom ports, annotation and readinessProbe?

Reply:
Done (I think...)